src/SourceConstraints.hs

```diff
@@ -122,8 +122,16 @@ warnings context@Context{..} (L sourceSpan node) =
     descend :: Data a => a -> WarningMessages
     descend = unionManyBags . gmapQ (descend `ext2Q` warnings context)
 
-locatedWarning :: Data a => Context -> a -> Maybe ErrMsg
-locatedWarning context = mkQ empty sortedImportStatement
+data IEClass = Module String | Type String | Operator String | Function String
```

Comment:
@mbj I'm curious if we have any rules regarding the canonical source location for type definitions. Is it "ok" to have this "interleaved" here with function definitions?

Reply:
Its only used here, and its not exported. So I added it just above the only usage side.

Reply:
That seems fair to me.